aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/NotificationClient.java

```diff
@@ -277,23 +277,44 @@ public String getChannelType() {
     }
 
     /**
-     * Handles pinpoint push messages by posting a local notification when
+     * Handles Pinpoint push messages by posting a local notification when
      * the app is in the background, or sending a local broadcast if the app is
      * in the foreground. Also on Api level 19 devices and above, if local
      * notifications have been disabled and the app is in the background, a
      * local broadcast is sent.
      *
      * @param notificationDetails the notification message received by the device's messaging service
-     * @return {@link NotificationClient.CampaignPushResult}.
+     * @return {@link PushResult}.
+     *
+     */
+    public PushResult handleNotificationReceived(final NotificationDetails notificationDetails) {
+        return notificationClientBase.handleNotificationReceived(notificationDetails);
+    }
+
+    /**
+     * @deprecated see {@link #handleNotificationReceived(NotificationDetails)}
      */
+    @Deprecated
     public CampaignPushResult handleCampaignPush(final NotificationDetails notificationDetails) {
-        return notificationClientBase.handleCampaignPush(notificationDetails);
+        return CampaignPushResult.valueOf(notificationClientBase.handleNotificationReceived(notificationDetails).toString());
     }
 
     /**
-     * Result values of handling a pinpoint push message.
+     * @deprecated see {@link PushResult}
      */
     public enum CampaignPushResult {
```

Comment:
The Javadoc `@deprecated` is good. But, it we should get a Java `@Deprecated` on the class itself, as well.

Reply:
That was a miss on my part. I'm adding that.